src/firefox/index.js

```diff
@@ -89,12 +99,54 @@ export function createProfile(
 }
 
 
+/*
+ * Copies an existing Firefox profile and creates a new temporary profile.
+ * It resolves with the new profile object.
+ *
+ * The temporary profile will be deleted when the system process exits.
+ *
+ * The existing profile can be specified as a directory path or a name of
+ * one that exists in the current user's Firefox directory.
+ */
+export function copyProfile(
+    profileDirectory: string,
+    {copyFromUserProfile=FirefoxProfile.copyFromUserProfile}: Object = {})
+    : Promise {
+
+  let copy = promisify(FirefoxProfile.copy);
+  let copyByName = promisify(copyFromUserProfile);
+
+  return isDirectory(profileDirectory)
+    .then((dirExists) => {
+      if (dirExists) {
```

Comment:
Currently this `copyProfile` helper doesn't set the preferences on the profile after copying it, as it is done in its `createProfile` counterpart.

We have to set the preferences even when we are cloning an existent profile (either by setting the preferences in this method or by refactoring out the "set the app preferences" part from the `createProfile` method and share it with both the methods that are used to create a profile) or the installed addon will probably not work correctly in the copied profile (e.g. the addon is disabled because is not signed and the copied preferences doesn't disable the signatures checks)